### PR TITLE
Bump libwally-core to 0.8.5

### PR DESCRIPTION
**Pull request by Sjors**
For some reason I can no longer build for the device or simulator on macOS 12.3.1:

```
Undefined symbols for architecture x86_64:
  "_secp256k1_nonce_function_default", referenced from:
      _wally_internal_ec_nonce_fn in libwallycore.a(libwallycore_la-internal.o)
...
```

### CODE REVIEW

**Comment by Sjors** [comment]
Building the latest master branch from scratch still works for me. I tried reverting https://github.com/ElementsProject/libwally-core/pull/32 but that actually made things worse. My guess is that something in https://github.com/ElementsProject/libwally-core/pull/321 requires updating our build stuff.

@jurvis any thoughts for what would be a quick fix? Or is it better to bite the bullet and make a more rigorous improvement like #61?

**Comment by jurvis** [comment]
TL;DR: #61 should hopefully fix a `libwally-core` build issue with `libtool@2.4.7`, support M1 macs running `arm64` simulator targets, support xcframeworks for potential Swift Package Manager integration in the future

@Sjors the improvement I'm working on in #61 hopefully achieves a couple of things, actually:
1. Update to `libwally-core@0.8.5` which fixes [build issues](https://github.com/Sjors/libwally-swift/issues/59) for me with `libtool@2.4.7`
2. Support running Libwally on M1 Mac running `arm64` simulators. This necessitates the use of `xcframeworks` afaik. `lipo` will end up packaging both simulator and device `arm64` targets which Xcode will not like 😬 
3. With that said, 1 enables us to actually clean up our podspec file, and may even mean we can delete the Carthage build script -- since we're actually building the `.xcframework` directly for users to drag into their project.

At this current moment, if you checkout https://github.com/Sjors/libwally-swift/pull/61/commits/60ed50a7c9f415166c413d3d215caf835ec43f75, run `./build-libwally.sh -sdc`, and drag `Libwally.xcframework` into any iOS project, it builds!

However, I'm having a hard time trying to debug getting it to work with Cocoapods ([sample project](https://github.com/jurvis/LibwallyCocoapodsSample)):
image

and Carthage currently spits out [a bunch of errors](https://gist.github.com/jurvis/0fb9847467ff5e0917f619d580f97d66) for me.
